src/benchmarks/micro/corefx/System.Text.RegularExpressions/Perf.Regex.Common.cs

```diff
@@ -13,35 +13,68 @@ namespace System.Text.RegularExpressions.Tests
     [BenchmarkCategory(Categories.CoreFX)]
     public class Perf_Regex_Common
     {
-        private const string EmailPattern = @"^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,12}|[0-9]{1,3})(\]?)$";
-        private const string DatePattern = @"\b(?<month>\d{1,2})/(?<day>\d{1,2})/(?<year>\d{2,4})\b";
-        private const string IPPattern = @"(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])";
-        private const string UriPattern = @"[\w]+://[^/\s?#]+[^\s?#]+(?:\?[^\s#]*)?(?:#[^\s]*)?";
-
         private Regex _email, _date, _ip, _uri;
+        private Regex _searchWord, _searchWords, _searchSet, _searchBoundary;
 
-        [Params(RegexOptions.None, RegexOptions.Compiled)]
+        [Params(RegexOptions.None, RegexOptions.Compiled, RegexOptions.Compiled | RegexOptions.IgnoreCase)]
         public RegexOptions Options { get; set; }
 
         [GlobalSetup]
         public void Setup()
         {
-            _email = new Regex(EmailPattern, Options);
-            _date = new Regex(DatePattern, Options);
-            _ip = new Regex(IPPattern, Options);
-            _uri = new Regex(UriPattern, Options);
+            _email = new Regex(@"^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,12}|[0-9]{1,3})(\]?)$", Options);
+            _date = new Regex(@"\b\d{1,2}\/\d{1,2}\/\d{2,4}\b", Options);
+            _ip = new Regex(@"(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])", Options);
+            _uri = new Regex(@"[\w]+://[^/\s?#]+[^\s?#]+(?:\?[^\s#]*)?(?:#[^\s]*)?", Options);
+
+            _searchWord = new Regex(@"tempus", Options);
+            _searchWords = new Regex(@"tempus|magna|semper", Options);
+            _searchSet = new Regex(@"\w{10,}", Options);
+            _searchBoundary = new Regex(@"\b\w{10,}\b", Options);
         }
 
-        [Benchmark] public void Email() => _email.IsMatch("yay.performance@dot.net");
-        [Benchmark] public void Date() => _date.IsMatch("Today is 11/18/2019");
-        [Benchmark] public void IP() => _ip.IsMatch("012.345.678.910");
-        [Benchmark] public void Uri() => _uri.IsMatch("http://example.org");
+        [Benchmark] public void Email_IsMatch() => _email.IsMatch("yay.performance@dot.net");
+        [Benchmark] public void Email_IsNotMatch() => _email.IsMatch("yay.performance@dot.net#");
```

Comment:
How big is the difference in the reported time for `IsMatch` and `IsNotMatch`? I wonder if the execution path is different enough to make it worth to "duplicate" the benchmarks (https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#Code-Paths)

Reply:
> Another disadvantage of adding the "IsNotMatch" test case is changing the IDs of the existing benchmarks

I don't think that's important.  We should ensure we have the benchmarks we want to have.  If we're not measuring the right things, we should stop doing so and measure the right things.

> How big is the difference in the reported time for IsMatch and IsNotMatch?

Big.

> I wonder if the execution path is different enough to make it worth to "duplicate" the benchmarks

It is.  The match case is the happy path: every check as part of the regex succeeds.  The not-match case highlights the quality of the regex engine, since that's where backtracking potentially kicks in.

> I wonder if instead of this we should have a single benchmark where the mismatched character is the first char.

That'd be a fine additional case, but it's not the most important case.